app/Insightly/InsightlyServiceProvider.php

```diff
@@ -27,7 +28,9 @@ public function register(): void
     {
         $this->app->bind(InsightlyMappingRepository::class, EloquentInsightlyMappingRepository::class);
 
-        $this->app->singleton(InsightlyClient::class, function () {
+        $this->app->bind(InsightlyClient::class, HttpInsightlyClient::class);
+
+        $this->app->singleton(HttpInsightlyClient::class, function () {
```

Comment:
What's the point of doing `bind()` and then `singleton()` with the implementation class name, if you could just do the `singleton()` definition with the interface name (like you did initially)?

Reply:
I personally prefer to just define the `singleton()` with the interface name, it's one less abstract layer that you need to resolve in your head when seeing how everything is wired. Only if the implementation can be auto-wired I'd use `bind()` because then you can just point out the implementation without implementing it yourself.

Reply:
Well, this was a merge conflict as both Luc and I made a change. I prefer the direct binding as well instead of adding a step, but maybe @LucWollants had his reasons?

Reply:
I think the issue was that the `HttpInsightlyClient` was indeed created as a singleton. But `InsightlyClient` gets injected everywhere. So when running the application I got an error because `InsightlyClient` could not be created.

I quickly fixed that by binding the interface `InsightlyClient` to  `HttpInsightlyClient`. But now I notice I could also have changed the singleton key from `HttpInsightlyClient` to `InsightlyClient`.

So fine for me to just use the singleton with `InsightlyClient`